src/Processors/Transforms/ArrayJoinTransform.cpp

```diff
@@ -1,32 +1,55 @@
 #include <Processors/Transforms/ArrayJoinTransform.h>
 #include <Interpreters/ArrayJoinAction.h>
+#include "Core/Field.h"
 
 namespace DB
 {
 
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
 Block ArrayJoinTransform::transformHeader(Block header, const ArrayJoinActionPtr & array_join)
 {
-    array_join->execute(header);
-    return header;
+    auto columns = header.getColumnsWithTypeAndName();
+    array_join->prepare(columns);
+    Block res{std::move(columns)};
+    res.setColumns(res.mutateColumns());
```

Comment:
why `res.setColumns(res.mutateColumns());` is needed?

Reply:
Make column not null, otherwise uts like tests/queries/0_stateless/00041_big_array_join.sql would failed